### PR TITLE
Added update for "AWS::EC2::Instance" CFn Resource Provider

### PR DESCRIPTION
**Pull request by Morijarti**
## Changes

During migration to new resource provider model, update method from GenericModel was skipped. 
This update reintroduces update

### CODE REVIEW

File: localstack/services/ec2/resource_providers/aws_ec2_instance.py [hunk 1]
```diff
@@ -303,4 +303,31 @@ def update(
 
 
         """
-        raise NotImplementedError
+        desired_state = request.desired_state
+        ec2 = request.aws_client_factory.ec2
+
+        groups = desired_state.get("SecurityGroups", desired_state.get("SecurityGroupIds"))
```
Comment (dominikschubert) [inline localstack/services/ec2/resource_providers/aws_ec2_instance.py:309]: That seems like a remnant from the way the old provider worked (fetch_state), but guess we can allow it for now

File: localstack/services/ec2/resource_providers/aws_ec2_instance.py [hunk 2]
```diff
@@ -303,4 +303,31 @@ def update(
...
+            InstanceId=desired_state["Id"],
+            InstanceType={"Value": desired_state["InstanceType"]},
+            **kwargs,
+        )
+
+        response = ec2.describe_instances(InstanceIds=[desired_state["Id"]])
+        instance = response["Reservations"][0]["Instances"][0]
+        if instance["State"]["Name"] != "running":
+            return ProgressEvent(
+                status=OperationStatus.desired_state,
```
Comment (dominikschubert) [inline localstack/services/ec2/resource_providers/aws_ec2_instance.py:324]: ?

**Review by dominikschubert — CHANGES_REQUESTED** [review]
Thanks for adding this back.

ProgressEvent status seems to be incorrect though

File: localstack/services/ec2/resource_providers/aws_ec2_instance.py (hunk 2 above)
Comment (Morijarti) [reply localstack/services/ec2/resource_providers/aws_ec2_instance.py:324]: oh dear, let me fix it right away

Comment (Morijarti) [reply localstack/services/ec2/resource_providers/aws_ec2_instance.py]: Fixed

**Review by dominikschubert — APPROVED** [review]
LGTM, thanks for fixing the regression

**Pull request merged**